Python full version in pipfile

jerempy opened the pull request:
Thank you for contributing to Pipenv!


### The issue
https://github.com/pypa/pipenv/issues/5345

A user requested that when the full version is specified that it gets added to a new pipfile when it is created so that he/she wouldn't need to manually add it themselves.


### The fix

Found the spot where the pipefile [requires] section gets created and modified it slightly to save based on if full version was specified or not.


### The checklist

* [x] Associated issue
* [x] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix.rst`, `.feature.rst`, `.behavior.rst`, `.doc.rst`. `.vendor.rst`. or `.trivial.rst` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

<!--
### If this is a patch to the `vendor` directory...

Please try to refrain from submitting patches directly to `vendor` or `patched`, but raise your issue to the upstream project instead, and inform Pipenv to upgrade when the upstream project accepts the fix.

A pull request to upgrade vendor packages is strongly discouraged, unless there is a very good reason (e.g. you need to test Pipenv’s integration to a new vendor feature). Pipenv audits and performs vendor upgrades regularly, generally before a new release is about to drop.

If your patch is not or cannot be accepted by upstream, but is essential to Pipenv (make sure to discuss this with maintainers!), please remember to attach a patch file in `tasks/vendoring/patched`, so this divergence from upstream can be recorded and replayed afterwards.
-->

oz123 reviewed (changes requested):
Thank you for your PR.

Please fix the linting issue, and add a news fragment.

This is then good to be merged!

oz123 commented:
Please rename the news snippet to:   5345.behavior.rst

oz123 commented:
As I suspected this change leads to breakage of CI.
```
Successfully set up CPython (3.7.14)

```

I think that pinning to a minor version is not a good thing. Unless it's additional. What I mean is, we should have both, e.g.:

```
python_version=3.7
python_full_verison=3.7.14
```

jerempy commented:
@oz123 I updated it so now the pipfile will record both the python version and the full version in addition if specified. Also updated tests so should work for whatever python version the test is running. Sorry for the back and forth and hopefully we good to go now 🤞

oz123 commented on news/5345.behavior.rst:
```diff
@@ -0,0 +1 @@
+New pipfiles show python_full_version under [requires] if specified
```
Can you please add a longer explanation here?

jerempy replied:
ok cool. good call. Hows it now?

oz123 replied:
Much better. Thank you.

matteius commented:
What happens if a new minor version is released?   Will users have to manually update the `python_full_version`?
